x-pack/plugins/security_solution/public/management/components/artifact_list_page/hooks/use_artifact_create_item.ts

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  CreateExceptionListItemSchema,
+  ExceptionListItemSchema,
+} from '@kbn/securitysolution-io-ts-list-types';
+import { useMutation } from 'react-query';
+import { HttpFetchError } from 'kibana/public';
+import { ExceptionsListApiClient } from '../../../services/exceptions_list/exceptions_list_api_client';
+
+// FIXME: delete entire file once PR# 125198 is merged. This entire file was copied from that pr
+
+export interface CallbackTypes {
+  onSuccess?: (updatedException: ExceptionListItemSchema) => void;
+  onError?: (error?: HttpFetchError) => void;
+  onSettled?: () => void;
+}
+
+export function useCreateArtifact(
```

Comment:
this can be removed in favour of the custom hooks

Reply:
I'll do this in my next PR, where I will also add tests + new prop for handling submit. is that cool?

Reply:
Sure! Sounds good to me!